### PR TITLE
make sure params --split and --merge are not specified at same time in gguf-split

### PR DESCRIPTION
**Pull request by kylo5aby**
- [x] I have read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md)
- Self-reported review complexity:
  - [X] Low
  - [ ] Medium
  - [ ] High

### CODE REVIEW

**Review by ggerganov — COMMENTED** [review]
These checks in the loop are hard to follow:

```c++
        if (is_op_set) {
            throw std::invalid_argument("error: either --split or --merge can be specified, but not both");
        }
        ...
        if (is_mode_set) {
            throw std::invalid_argument("error: either --split-max-tensors or --split-max-size can be specified, but not both");
        }
```

Seems something wrong with them already, or at the very least will cause trouble in the future if we add more args. Better to fix this properly.

**Comment by ngxson** [comment]
I'd suggest adding `SPLIT_OP_NONE` and `enum split_mode` for it (default to `SPLIT_MODE_NONE`).

> will cause trouble in the future if we add more args.

For `gguf-split`, I don't think there will be many args added in the future, as this example is quite basic. This check is pretty much optional though, it's there for better user experience. One way is to remove it altogether if it causes trouble.

**Comment by kylo5aby** [comment]
> I'd suggest adding `SPLIT_OP_NONE` and `enum split_mode` for it (default to `SPLIT_MODE_NONE`).
> 
> > will cause trouble in the future if we add more args.
> 
> For `gguf-split`, I don't think there will be many args added in the future, as this example is quite basic. This check is pretty much optional though, it's there for better user experience. One way is to remove it altogether if it causes trouble.

yes, I think these enums will be helpful for handling repeated parameters, which is better for user experience

**Review by ngxson — APPROVED** [review]

**Review by ggerganov — APPROVED** [review]

File: examples/gguf-split/gguf-split.cpp [hunk 1]
```diff
@@ -265,13 +275,15 @@ struct split_strategy {
...
+        if (params.mode == MODE_SIZE) {
             // split by max size per file
             return next_size > params.n_bytes_split;
-        } else {
+        } else if (params.mode == MODE_TENSOR) {
             // split by number of tensors per file
             return i_tensor > 0 && i_tensor < n_tensors && i_tensor % params.n_split_tensors == 0;
         }
+        // should never happen
+        return false;
```
Comment (slaren) [inline examples/gguf-split/gguf-split.cpp:286]: If this can never happen, then make this a `GGML_ABORT` instead.
```suggestion
        // should never happen
        GGML_ABORT("invalid mode");
```

**Review by slaren — APPROVED** [review]
